### PR TITLE
Change offer model to support contact options

### PR DESCRIPTION
**Pull request by Es7evam**
Changed several filed to support the option of the offer being attached with contact information.
The user now can choose to display or not his telephone and/or email. In case the option is to not display then the field will come empty when doing a get request.

models/offer.go | Add telephone and email as strings and their representation at json file.

migrations/3_offer_table.up.sql | Add telephone and email as text. 

handler/offerHandler.go | Add telephone and email as bool fields at offerInsertion | Add new fields at test's json.

usecases/offerCases.go | Add userDAL, InsertOffer modified to deal with the new parameters of telephone and email and call the DAL function as before.

tools/test.go | CreateFakeOffer sets dummie telephone and email so that all of the other tests won't need to be modified.

docs/API.md | Modify Offer GET and POST API to adapt to new fields. Also modify Get User Offers
            
dal/offerDAL.go | Add telephone and email attributes at the insertion query.

### CODE REVIEW

File: backend/docs/API.md [hunk 1]
```diff
@@ -172,7 +172,9 @@ For `size` = 2
       "minprice": 30,
       "userid": 1,
       "categoryid": 2,
-      "timestamp": "2019-06-13T10:28:42.411752Z"
+      "timestamp": "2019-06-13T10:28:42.411752Z",
+      "telephone": "0000-0000",
```
Comment (mbotarro) [inline backend/docs/API.md]: Could you please add the expiration time in these examples too? Thanks for the help!

File: backend/usecases/offerCases.go [hunk 2]
```diff
@@ -28,6 +30,23 @@ func (rc *OfferController) GetLastOffers(before time.Time, size int) ([]models.O
 
 // InsertOffer inserts the given offer into the database, calling the DAL package function.
 // It returns error != nil in case some error occured.
-func (rc *OfferController) InsertOffer(offer models.Offer) error {
+func (rc *OfferController) InsertOffer(offer models.Offer, telephone bool, email bool) error {
+	u, err := rc.userDAL.GetUserInfo(offer.Userid)
+
```
Comment (mbotarro) [inline backend/usecases/offerCases.go]: nit: usually we don't use a space between error assignment and check.

File: backend/docs/API.md (hunk 1 above)
Comment (Es7evam) [reply backend/docs/API.md]: Did it in commit 7e6c693

File: backend/usecases/offerCases.go (hunk 2 above)
Comment (Es7evam) [reply backend/usecases/offerCases.go]: Corrected in 7e6c693

**Pull request merged**